File: src/snippets/ip-threat-intelligence/shared-py/InspectDecision.py
# ... imports, client configuration, etc
# See https://docs.arcjet.com/get-started
decision = await aj.protect(request)

if decision.is_denied():
    # A filter or another rule already denied this request.
    return JSONResponse({"error": "Forbidden"}, status_code=403)

ip = decision.ip

if ip.is_hosting():
    # Hosting and data center IPs are often automated clients.
    pass

if ip.is_vpn() or ip.is_proxy() or ip.is_tor():
    # Apply your policy for anonymized traffic.
    pass

if ip.is_abuser():
    # The IP is associated with known abuse.
    pass

if ip.is_relay() and decision.ip_details and decision.ip_details.service == "Apple Private Relay":
    # Apple Private Relay requires a paid iCloud subscription.
    pass

details = decision.ip_details
threat = details.threat if details else None

if threat and not threat.is_safe and threat.risk_level == "critical":
    # High-risk IP. threat.activities may include brute_force, scanning, or
    # botnet. threat.network_types may include hosting, vpn, proxy, or tor.
    pass

File: src/snippets/ip-threat-intelligence/shared-js/InspectDecision.ts
// ... imports, client configuration, etc
// See https://docs.arcjet.com/get-started
const decision = await aj.protect(req);

if (decision.isDenied()) {
  // A filter or another rule already denied this request.
  return new Response("Forbidden", { status: 403 });
}

const ip = decision.ip;

if (ip.isHosting()) {
  // Hosting and data center IPs are often automated clients.
}

if (ip.isVpn() || ip.isProxy() || ip.isTor()) {
  // Apply your policy for anonymized traffic.
}

if (ip.isAbuser()) {
  // The IP is associated with known abuse.
}

if (ip.isRelay() && ip.hasService() && ip.service === "Apple Private Relay") {
  // Apple Private Relay requires a paid iCloud subscription.
}

const threat = ip.threat;

if (threat && !threat.isSafe && threat.riskLevel === "critical") {
  // High-risk IP. threat.activities may include brute_force, scanning, or
  // botnet. threat.networkTypes may include hosting, vpn, proxy, or tor.
}
